### ISSUE TITLE
High level REST client support for cluster/indices administration

### ISSUE THREAD

**Issue by sherry-ger** [elastic/elasticsearch#25847]
As the hight level REST client can be upgraded independent of the Elasticsearch cluster, it is much friendlier than the transport client.  Currently, it does not support cluster and indices administration APIs as the transport client does.  It would be nice to have the ability to create/update/delete indices/templates using the HL REST client too.

**Comment by jeacott1** [comment]
it would be nice if the transport client  just used rest under the covers and nobody really cared about the mechanics.

**Comment by javanna** [comment]
> it would be nice if the transport client just used rest under the covers and nobody really cared about the mechanics.

hi @jeacott1 that would be great for users, for sure. But it would imply that we need to have implemented all of the methods going through REST, meaning a lot more work needed before we can  even start having users go through http. Also, it would mean legacy methods that are already 5+ years old and we want to get rid of would need to be implemented. We decided to start a new client that's as close as possible to the transport client, which tries to make it easy for users to upgrade, while allowing us to clean stuff up at the same time. I hope this sounds reasonable :)

**Comment by javanna** [comment]
We do want to add support for these methods, the end goal is possibly feature parity with the Transport Client. Delete Index API has recently been added. Pull requests are very welcome! We are tracking API completeness of the high-level REST client as part of #27205 , hence I am closing this issue in favour of #27205 .
